vispy/app/backends/_glfw.py

```diff
@@ -447,7 +449,18 @@ def _on_key_press(self, _id, key, scancode, action, mod):
         else:
             return
         self._process_mod(key, down=down)
-        fun(key=key, text=text, modifiers=self._mod)
+        
+        # NOTE: GLFW only provides localized characters via _on_key_char, so if
+        # this event contains a character we store all other data and dispatch
+        # it once the final unicode character is sent shortly after.
+        if text != '' and action == glfw.GLFW_PRESS:
+            self._next_key_events.append((fun, key, self._mod))
+        else:
+            fun(key=key, text=text, modifiers=self._mod)
+
+    def _on_key_char(self, _id, text):
+        (fun, key, mod) = self._next_key_events.pop(0)
+        fun(key=key, text=chr(text), modifiers=mod)
```

Comment:
@almarklein No it's not intended, and when testing it doesn't behave that way here. Did I miss something? :-)

That code just delays all key press events which have a character associated with them (storing them in a list), then when GLFW sends a character event it will use the correct unicode character and dispatch the event shortly after.

One bug I've just noticed is that the `text` for GLFW_PRESS will be locale-correct but for GLFW_RELEASE it will not be...